### ISSUE TITLE
Org admins cannot invite users in 9.0.x

### ISSUE THREAD

**Issue by rda0** [grafana/grafana#51634]
**What happened**:
Organization administrators (using the default basic org admin role) can no longer invite new users since 9.0.x. The button in the UI is missing:
![grafana_org_admin_button_missing](https://user-images.githubusercontent.com/6731714/176669518-2fec18e1-aa6b-4616-9c2f-e8db98a14137.png)

Using the API: `curl -H "Authorization: Bearer **SECRET KEY**=" https://grafana.phys.ethz.ch/api/org/invites` shows the error:

```
Permissions needed: users:create
```

**What you expected to happen**:

This used to work with version 8:
![grafana_org_admin_button_there](https://user-images.githubusercontent.com/6731714/176669954-c459e5eb-e7e1-42e1-9e5e-7ea9c8356f0d.png)

**How to reproduce it (as minimally and precisely as possible)**:

Upgrade to v9.

**Anything else we need to know?**:

The docs at https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#organization-roles state that this permission should be the default (`Add and edit users`):

Permission | Organization administrator | Editor | Viewer
-- | -- | -- | --
View dashboards | x | x | x
Add, edit, delete dashboards | x | x |  
Add, edit, delete folders | x | x |  
View playlists | x | x | x
Add, edit, delete playlists | x | x |  
Create library panels | x | x |  
View annotations | x | x | x
Add, edit, delete annotations | x | x |  
Access Explore | x | x |  
Add, edit, delete data sources | x |   |  
Add and edit users | x |   |  
Add and edit teams | x |   |  
Change organizations settings | x |   |  
Change team settings | x |   |  
Configure application plugins | x

**Environment**: 
- Grafana version: 9.0.2
- Data source type & version: 
- OS Grafana is installed on: Debian 10 Buster
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by joshhunt** [comment]
Can reproduce - Org Admin cannot create new users, which is at least inconsistent with documentation.

**Comment by stakeconomy** [comment]
We have the same issue. Would love to see this returned

**Comment by Lusitaniae** [comment]
```Grafana Teams API answered with HTTP 500```

**Comment by Lusitaniae** [comment]
Work around:
- Login as Grafana Admin (not an organization admin) 
- Invite users as before
- and/or assign your org admins with Grafana Admin permissions

**Comment by SockenSalat** [comment]
@Lusitaniae even better work-around: do not install Grafana 9 at all

**Comment by IevaVasiljeva** [comment]
Thanks for reporting the issue! A fix for it has been merged in https://github.com/grafana/grafana/pull/51668 and will be released with Grafana v9.0.3.

Please note that this fix only allows Organization Admins to add already existing users to their organisation, but doesn't allow them to add new users to Grafana. In order to allow someone to add new users to Grafana, you will have to give them Grafana Server Admin role.

Hope this resolves the problem!

**Comment by rda0** [comment]
@IevaVasiljeva Thank you for your answer.

> Please note that this fix only allows Organization Admins to add already existing users to their organisation, but doesn't allow them to add new users to Grafana. In order to allow someone to add new users to Grafana, you will have to give them Grafana Server Admin role.

We do not need this functionality in our case, since we manage organizations via LDAP group mappings. And from a security perspective, it is unwanted that org admins have server admin rights.

> Hope this resolves the problem!

No not exactly. Previously our org admins were able to invite external users (which are not already in Grafana or in LDAP) using this invite feature. This somehow generated an invitation email where external users were able to sign up as an org member. This was a nice feature for some of our org admins, when they had external users who needed to collaborate on their projects.

If this feature will be off now by default, could you please instruct what exact permissions need to be changed in the [existing basic roles](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/rbac-fixed-basic-role-definitions/) to get the behavior from before?

**Comment by IevaVasiljeva** [comment]
@rda0 to allow Organization Admins to create new users, you would need to add `users:create` permission to Admin basic role.

I know that this is a behaviour change, but we talked it through internally and decided that user creation falls under Grafana Server Admin domain, and that we should not grant it to Organization Admins by default. But of course we're open to feedback.

**Comment by SockenSalat** [comment]
@IevaVasiljeva 

I think this is just fine. But to mention this prominently unter the `breaking changes` section for the 9.0.0 release surely would have caused half the surprise. :thinking:

**Comment by IevaVasiljeva** [comment]
@SockenSalat that's a fair point, we did miss that in the release notes. We'll try to be more thorough job with the next release.

**Comment by ranryl** [comment]
I have same issue use version 9.0.3 (2022-07-14)

**Comment by vtorosyan** [comment]
Unfortunately we have missed the backport for the previous patch releases and the issue still exists for 9.0.3 and 9.0.4, sorry about that. Once we have got [this backport PR merged](https://github.com/grafana/grafana/pull/52553), the fix will go out with the next patch.

**Comment by IevaVasiljeva** [comment]
Hi all! Really sorry about missing the backport on this one! And thanks for fixing it @vtorosyan! A good lesson for not working on too many things at once 🙈

**Comment by rda0** [comment]
> @rda0 to allow Organization Admins to create new users, you would need to add `users:create` permission to Admin basic role.

I tried to configure this but nothing seems to work. I created the file `/etc/grafana/provisioning/access-control/basic_roles.yaml`:

```yaml
apiVersion: 2

roles:
  - name: 'basic:grafana_admin'
    global: true
    version: 3
    from:
      - name: 'basic:grafana_admin'
        global: true
    permissions:
      # Permissions to add
      - action: 'users:create'
```

Then restart the service, in the logs I see:

```
logger=settings t=2022-07-27T15:56:46.304193323+02:00 level=info msg="Path Provisioning" path=/etc/grafana/provisioning 
logger=settings t=2022-07-27T15:56:46.304259077+02:00 level=info msg="App mode production" 
...
```

The documentation states that provisioning needs to be enabled, but how? Or what other steps are needed?

Most internal links in the documentation seem to be broken (pointing to the page itself).

Reloading via API returns 404:

```
+rda:~$ http -a <admin-uid> POST https://example.com/api/admin/provisioning/access-control/reload
http: password for admin-uid@example.com: 
HTTP/1.1 404 Not Found
```

**Comment by vtorosyan** [comment]
@rda0 thanks for getting back and for the feedback.

**About Org Admins not being able to create new users**

As [mentioned above](https://github.com/grafana/grafana/issues/51634#issuecomment-1178903701), we have removed the ability for Org Admins to create new users, as we thought about it as a Grafana Server Admin function (and missing to communicate it as a breaking change was a mistake, sorry about that).

Based on the feedback we have received (in this issue and in other discussions), we realised that it was a wrong decision to remove that functionality. **We will be working on a fix now and release it with the next patch, which will revert the behavior to what it was before 9.0.x.**

**About using RBAC and documentation**

Regarding provisioning new roles/permissions, you are having issues since the [RBAC feature is only available in Grafana Enterprise](https://grafana.com/docs/grafana/next/administration/roles-and-permissions/#grafana-enterprise-user-permissions-features), sorry for the confusion. It is definitely confusing and not clear in the documentation, I have opened a PR to [discuss this](https://github.com/grafana/grafana/pull/52890). 

For more context about the documentation, we are trying to consolidate documentation to make it easier for readers to understand the feature set that Grafana offers more holistically, and with this, we have moved the RBAC documentation to a general section under [Roles and permissions](https://grafana.com/docs/grafana/next/administration/roles-and-permissions/), as a result now the RBAC docs are missing statements that it is available only for Grafana Enterprise. 

The broken links are now fixed with [the next version of docs](https://grafana.com/docs/grafana/next/administration/roles-and-permissions/access-control/rbac-provisioning/).

**Comment by IevaVasiljeva** [comment]
Here's a PR that reverts the functionality back to what it was before 9.0 - Org Admins will be able to invite new and existing users to the org: https://github.com/grafana/grafana/pull/52894

We're hoping to release this change with 9.0.6 and 9.1 beta next week.

Sorry for all the back and forth, and hopefully the solution works for you!

**Comment by BapRx** [comment]
I believe I have the same issue on Gravana `v8.5.11`: https://github.com/grafana/grafana/blob/v8.5.11/pkg/api/org_invite.go#L73

The change was introduced in this PR: https://github.com/grafana/grafana/pull/51964/files

I'm rollbacking to `v8.5.9` for now but @IevaVasiljeva can you revert this PR also?

Thank you

**Comment by IevaVasiljeva** [comment]
Hi @BapRx,

here's a backport for `8.5.x`: https://github.com/grafana/grafana/pull/55585

Just to set the expectations here - we don't do regular patch releases for Grafana 8 anymore, so it might take a while for the change to be released.

**Comment by johndunderhill** [comment]
@IevaVasiljeva This has resurfaced in Grafana v10.0.1 or earlier (I'm on Cloud).  The Invite button is missing and I cannot add oridinary Grafana users.  We do have login enabled, and we have existing ordinary (non-oAuth) users who can log in.  We've been doing it this way for some time.  We *cannot* add users via Grafana.com--this gives them visibility into the entire stack, as well as our API keys (!) and our billing information (!), even at the lowest permission level (Viewer).  All we want is to add some users who can view dashboards.  This has us at a standstill.  We don't understand why you would prevent us from creating new users.  This makes absolutely no sense.  If Enterprise admins want to lock this functionality out for their own security reasons, why not give them a simple flag for that, rather than trying to figure out what to do from other configuration settings?  Obviously that didn't work here.

**Comment by IevaVasiljeva** [comment]
Hi @johndunderhill! Thanks for reporting the issue and providing a clear description of your use case. And sorry for all the frustrations that removals of the `invite` button have caused you.

For context - the recent issues are not related to the changes I made and reverted a year ago (https://github.com/grafana/grafana/pull/52894), but come from us working on improving the integration between hosted Grafana instances and Grafana Cloud portal (https://github.com/grafana/grafana/pull/67031 and https://github.com/grafana/grafana/pull/68991). Unfortunately, while doing that we overlooked some potential instance configuration. We're aiming to release a fix in the next Grafana release. I'll link a PR here once it's ready.
